Description: Use execute a VACUUM operation on activity DB on startups every 10 days
Author: Marco Trevisan <marco.trevisan@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/919801
Forwarded: https://irclogs.ubuntu.com/2016/02/24/%23ubuntu-desktop.html#t10:44
Last-Update: 2021-01-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/data/org.gnome.zeitgeist.Engine.service.in
+++ b/data/org.gnome.zeitgeist.Engine.service.in
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=org.gnome.zeitgeist.Engine
-Exec=@bindir@/zeitgeist-daemon
+Exec=/bin/sh -c "@pkglibexecdir@/zeitgeist-maybe-vacuum; @bindir@/zeitgeist-daemon"
 SystemdService=zeitgeist.service
--- a/data/zeitgeist.service.in
+++ b/data/zeitgeist.service.in
@@ -5,5 +5,6 @@
 [Service]
 Type=dbus
 BusName=org.gnome.zeitgeist.Engine
+ExecStartPre=-@pkglibexecdir@/zeitgeist-maybe-vacuum
 ExecStart=@bindir@/zeitgeist-daemon
 Slice=session.slice
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -38,7 +38,7 @@
 	$(AM_V_GEN)$(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
 
 %.service: %.service.in
-	$(AM_V_GEN)sed -e "s|@bindir[@]|$(bindir)|" $< > $@
+	$(AM_V_GEN)sed -e "s|@bindir[@]|$(bindir)|; s|@pkglibexecdir[@]|$(pkglibexecdir)|" $< > $@
 
 clean:
 	rm -rf *.pyc *.~[0-9]~
